backend/src/main/java/ba/unsa/etf/si/bbqms/ws/controllers/TenantController.java

```diff
@@ -0,0 +1,32 @@
+package ba.unsa.etf.si.bbqms.ws.controllers;
+
+import ba.unsa.etf.si.bbqms.tenant_service.api.TenantService;
+import ba.unsa.etf.si.bbqms.ws.models.AddTenantDto;
+import ba.unsa.etf.si.bbqms.ws.models.ErrorResponseDto;
+import jakarta.persistence.EntityNotFoundException;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/api/v1/tenants")
+public class TenantController {
+    private final TenantService tenantService;
+
+    public TenantController(final TenantService tenantService) {
+        this.tenantService = tenantService;
+    }
+
+    @PostMapping()
+    public ResponseEntity addTenant(@RequestBody final AddTenantDto tenantDto) {
+        return ResponseEntity.ok().body(tenantService.addTenant(tenantDto));
+    }
+
+    @GetMapping("/{code}")
+    public ResponseEntity getTenantByCode(@PathVariable(name = "code") final String code) {
+        try {
+            return ResponseEntity.ok().body(tenantService.findByCode(code));
+        } catch (EntityNotFoundException e) {
+            return ResponseEntity.badRequest().body(new ErrorResponseDto(e.getMessage()));
```

Comment:
We shouldn't return the exception to users of the api. We should return an appropriate message.

`"Could not retrieve the tenant with code: " + code` should do the trick.

Reply:
The exception is instantiated here along with a similar message:

https://github.com/BBQMS2024/BBQMS/blob/b8541a5ccf8f3b4f5368672d579bd80c76648c92/backend/src/main/java/ba/unsa/etf/si/bbqms/tenant_service/implementation/TenantServiceImpl.java#L42-L45

Only the `ErrorResponseDto` which contains the message is returned to the API user.

Reply:
Fair enough.